pilot/pkg/serviceregistry/kube/controller/ambient/services.go

```diff
@@ -82,6 +137,68 @@ func (a *index) ServicesCollection(
 	return WorkloadServices
 }
 
+func (a *index) cleanUpServiceEntries(serviceEntries krt.Collection[*networkingclient.ServiceEntry],
+	kubeServices krt.Collection[*v1.Service],
+	opts krt.OptionsBuilder,
+) krt.Collection[*networkingclient.ServiceEntry] {
+	serviceEntriesByNamespacedHostname := krt.NewIndex(serviceEntries, "serviceEntriesByNamespacedHostname", func(se *networkingclient.ServiceEntry) []string {
```

Comment:
Dumb question: where do we ensure that service entries don't duplicate hosts across themselves?